apis/compute/v1alpha1/types.go

```diff
@@ -151,3 +151,127 @@ func (kc *KubernetesClusterClassList) GetPortableClassItems() []resource.Portabl
 	}
 	return items
 }
+
+// VirtualMachineSpec specifies the desired state of a VirtualMachine.
+type VirtualMachineSpec struct {
+	runtimev1alpha1.ResourceClaimSpec `json:",inline"`
+}
+
+var _ resource.Claim = &VirtualMachine{}
+
+// +kubebuilder:object:root=true
+
+// A VirtualMachine is a portable resource claim that may be satisfied by
+// binding to a Virtual Machine managed resource such as an AWS EC2 instance
+// or an Azure VM.
```

Comment:
I recall that the initial driver for this change was adding support for Packet, who market as bare metal. If we want to bind this kind to bare metal machines (either Packet or eventually on prem) perhaps `Machine` would be appropriate? This would fit both virtual and bare metal machines.

We might also consider `Instance`, which would have symmetry with the existing `CloudSQLInstance`, `RDSInstance`, etc types which tend to mean "a server running software X", but would probably make less sense for a bare metal machine.

Reply:
Agreed, it seems wrong to call it a VM if binding to bare metal. I would personally be in support of `Instance` because of the generality of the term. Packer refers to their bare metal machines as `Devices`.

Reply:
Just `Instance` seems very generic to me, especially when we have resulted derivatives like `InstanceController`. Maybe something more like `ComputeInstance`, `MachineInstance` or `CPUInstance`?

Reply:
@soorena776 looking back at this I would tend to agree with you. @prasek @bassam any thoughts on these options?

Reply:
I'd prefer `ComputeInstance` over `Instance`. I like `Machine` too, and would probably pick that in a vacuum, but `ComputeInstance` matches our other `FooInstance` claim names.

Reply:
MachineInstance  👍 

You’re getting a machine (VM or bare-metal). Compute is too general IMO.